standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -3236,18 +3238,28 @@ public NotificationEventResponse getNextNotification(long lastEventId, int maxEv
     NotificationEventResponse filtered = new NotificationEventResponse();
     if (rsp != null && rsp.getEvents() != null) {
       long nextEventId = lastEventId + 1;
+      long prevEventId = lastEventId;
       for (NotificationEvent e : rsp.getEvents()) {
+        LOG.debug("Got event with id : " + e.getEventId());
         if (e.getEventId() != nextEventId) {
-          LOG.error("Requested events are found missing in NOTIFICATION_LOG table. Expected: {}, Actual: {}. "
-                  + "Probably, cleaner would've cleaned it up. "
-                  + "Try setting higher value for hive.metastore.event.db.listener.timetolive. "
-                  + "Also, bootstrap the system again to get back the consistent replicated state.",
-                  nextEventId, e.getEventId());
-          throw new IllegalStateException(REPL_EVENTS_MISSING_IN_METASTORE);
+          if (e.getEventId() == prevEventId) {
+            LOG.error("NOTIFICATION_LOG table has multiple events with the same event Id {}. " +
+                    "Something went wrong when inserting notification events.  Bootstrap the system " +
+                    "again to get back teh consistent replicated state.", prevEventId);
```

Comment:
i am not sure if we need to handle this separately. As per code, this can never happen now.

Reply:
If it happens by any chance, say because of a db error, the error reported is quite misleading i.e. the cleaner has not removed any events in that case. So, I think it's better to differentiate between these two cases. If every duplicate eventIds show up at a proper error message will be conveyed.